### ISSUE TITLE
TrackerEvent.<init>  - ConcurrentModificationException

### ISSUE THREAD

**Issue by ShivaS0111** [snowplow/snowplow-android-tracker#692]
**Describe the bug**
Fatal Exception: java.util.ConcurrentModificationException

**To Reproduce**
unable to reproduce in dev environment, facing issue in prod, reported in the firebase crashlog.

**Expected behavior**
should not throw an error

**Screenshots**
<img width="878" alt="Screenshot 2024-06-18 at 2 14 29 PM" src="https://github.com/snowplow/snowplow-android-tracker/assets/48886501/7a3b7f88-a1eb-49ca-9576-83aee9b828d8">
<img width="898" alt="Screenshot 2024-06-18 at 2 15 38 PM" src="https://github.com/snowplow/snowplow-android-tracker/assets/48886501/9d49d48b-fb8c-4b89-86c0-a274ae873320">

**log file**
[snowplow_crash_issue_86b88a0970251b5bc3b1863997677833_crash_session_66713D77028800011DFB8A366C99B6D9_DNE_0_v2_stacktrace.txt](https://github.com/user-attachments/files/15883121/snowplow_crash_issue_86b88a0970251b5bc3b1863997677833_crash_session_66713D77028800011DFB8A366C99B6D9_DNE_0_v2_stacktrace.txt)

**Device informatoin (please complete the following information):**
 - Samsung Galaxy S9
 - OS: Android 10

 - Google Pixel 8 Pro
 - OS: Android 10
 
 - LGE Velvet
 - OS: Android 14
 
 - Huawei Y9 Prime 2019
 - OS: Android 10
 
 - Huawei P30 Pro
 - OS: Android 10
 
 - Xiaomi Redmi Note 10 5G
 - OS: Android 13

**Comment by jbeemster** [comment]
Hi @ShivaS0111 which version of the Tracker are you using?

**Comment by ShivaS0111** [comment]
HI @jbeemster, Snowplow tracker version: 4.1.2

**Comment by ShivaS0111** [comment]
Hi @jbeemster, any update on this, need to upgrade to any specific version?

**Comment by matus-tomlein** [comment]
Hi @ShivaS0111, thanks for getting back to us on the tracker version!

Unfortunately, we believe the issue is still present in the latest version of the tracker. We have added this to our backlog and will try to address in our upcoming sprints.

**Comment by avinash-ivy** [comment]
Hi @matus-tomlein 
Is there an ETA you can provide? We have decided to use Snowplow for remote logging. However, after pushing our changes live, we were forced to disable this feature due to crashes, so remote logging is currently unavailable for our app. Based on your ETA, we will decide whether we need to search for an alternative solution.

**Comment by matus-tomlein** [comment]
Sorry for the delay @avinash-ivy, this has been added to our next sprint, so should be released in around 2 weeks.

**Comment by matus-tomlein** [comment]
@avinash-ivy We are currently working on this and just to help us replicate – could you please let us know if at any point in your app, do you make changes to events that are already tracked? What I mean is that if you initialize an event (e.g.. using the `SelfDescribing` class), then you track it using `tracker.track(event)`, is there a chance in parallel with this or after, the event instance is being modified?
